### PR TITLE
Refactor PlantChillers

### LINKED ISSUE #7680: Electric Chiller Forgets to Update Sometimes
Issue overview
--------------
Possible related or duplicates:
 - #7134 
 - #6404

While refactoring the electric chiller for PlantComponent in #7637, a bunch of time was ~~spent~~ wasted debugging stray diffs in CompSetPtControl.idf.  As it turns out, even in develop that chiller model is misbehaving.  Primarily due to forgetting to update a number of what used to be module-level shared variables.  Overall, now that the chiller is refactored and cleaned up, someone needs to take a full pass to clean up all return paths to ensure variables are initialized/cleared wherever needed.  This will indeed cause diffs, so this needs to be done in a dedicated branch, not in conjunction with larger work.

This can be seen by just looking at the CompSetPtControl.idf outputs, especially how the condenser flow and delta T do not agree for some timesteps.

(FYI @mjwitte @mitchute)

### PR DESCRIPTION
**Pull request by mitchute**
Addresses #7680 

This was refactored once already in #7637, but this should be a more thorough refactor of this to make sure all control paths are correct and that variables get set/reset as needed.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by nrel-bot-2** [comment]
@mitchute it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@mitchute @lgentile it has been 28 days since this pull request was last updated.

**Comment by mitchute** [comment]
Ready for review here.

File: src/EnergyPlus/FaultsManager.cc [hunk 1]
```diff
@@ -545,12 +545,6 @@ namespace FaultsManager {
                 int ChillerNum;
 
                 if (UtilityRoutines::SameString(SELECT_CASE_VAR, "Chiller:Electric")) {
-                    // Read in chiller if not done yet
-                    if (PlantChillers::GetElectricInput) {
-                        PlantChillers::ElectricChillerSpecs::getInput();
-                        PlantChillers::GetElectricInput = false;
-                    }
-
```
Comment (Myoldmopar) [inline src/EnergyPlus/FaultsManager.cc:553]: Nice cleanup here, no need to check this now.

File: src/EnergyPlus/FaultsManager.cc [hunk 2]
```diff
@@ -566,7 +560,7 @@ namespace FaultsManager {
                         ErrorsFound = true;
                     } else {
 
-                        if (PlantChillers::ElectricChiller(ChillerNum).CondenserType != PlantChillers::WaterCooled) {
+                        if (PlantChillers::ElectricChiller(ChillerNum).CondenserType != PlantChillers::CondType::WaterCooled) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/FaultsManager.cc:563]: I smell a scoped enum, nice.

**Review by Myoldmopar — APPROVED** [review]
This looks great.  So many `using`s cleaned out, variables rescoped and cleaned up, args cleaned up.  Good stuff!

**Comment by Myoldmopar** [comment]
CI is all clean (Mac has been funny so I'm not waiting on it), and this branch is 26 commits ahead of develop, not behind.  Merging, thanks @mitchute.

**Pull request merged**